### PR TITLE
create file with 600 permission

### LINKED ISSUE #908: Generated private keys are world-readable by default
Make sure the file permissions in the CreateConfig command are 600

### PR DESCRIPTION
**Pull request by vedhavyas**
Closes #908

### CODE REVIEW

File: utils/io.go [hunk 1]
```diff
@@ -10,19 +10,22 @@ import (
 
 // WriteKeyToPemFile writes encode of key and purpose to the file
 func WriteKeyToPemFile(fileName string, keyPurpose string, key []byte) error {
```
Comment (mikiquantum) [inline utils/io.go]: Can we extend the existing `io_test.go` to check that the file permissions are the expected ones?

File: utils/io.go [hunk 2]
```diff
@@ -10,19 +10,22 @@ import (
...
-	f, err := os.Create(fileName)
+	// we are going with 0622 so that when umask is applied, we will get 0600 file permission
+	f, err := os.OpenFile(fileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0622)
```
Comment (mikiquantum) [inline utils/io.go]: Does it work the same across OSs?

Comment (vedhavyas) [reply utils/io.go]: There is no clear way of confirming this. The default Umask is 022 for most OS's. If someone change the umask to be more strict, then we will have problems with current(0644) or old (0666) as well. So I guess, it is okay

Comment (mikiquantum) [reply utils/io.go]: wouldnt it be safer to create file first and then change the permissions? That way we overcome possible restrictive values of Umask?

Comment (vedhavyas) [reply utils/io.go]: I think it should work

File: utils/io.go [hunk 3]
```diff
@@ -15,15 +15,16 @@ func WriteKeyToPemFile(fileName string, keyPurpose string, key []byte) error {
 		return err
 	}
 
+	defer f.Close()
```
Comment (vimukthi-git) [inline utils/io.go]: is there no way to handle error here?

File: utils/io.go [hunk 4]
```diff
@@ -15,15 +15,16 @@ func WriteKeyToPemFile(fileName string, keyPurpose string, key []byte) error {
...
 		Type:  keyPurpose,
 		Bytes: key,
 	}
 	if err := pem.Encode(f, block); err != nil {
 		return err
 	}
-	f.Close()
-	return nil
+
+	return f.Chmod(os.FileMode(0600))
```
Comment (vimukthi-git) [inline utils/io.go]: is there no standard constant defined for this?

Comment (vedhavyas) [reply utils/io.go]: There will be million combinations if they go with constants ;)

Comment (vimukthi-git) [reply utils/io.go]: haha but there are bit masks that can be used right?

Comment (vimukthi-git) [reply utils/io.go]: not critical ;)

Comment (vedhavyas) [reply utils/io.go]: If you mean by using the standard perm 0777 and masking it. I think we will lose readability there. Would prefer keeping it as `0600`

**Pull request merged**